osutil/exec.go

```diff
@@ -162,6 +162,26 @@ func KillProcessGroup(cmd *exec.Cmd) error {
 	return syscallKill(-pgid, syscall.SIGKILL)
 }
 
+// truncateOutput truncates input data by maxLines, imposing maxBytes limit (total) for them.
+func truncateOutput(data []byte, maxLines, maxBytes int) bytes.Buffer {
+	truncdata := bytes.Buffer{}
+	if maxBytes > len(data) {
+		maxBytes = len(data)
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(data[:maxBytes]))
+	for i := 0; scanner.Scan(); i++ {
+		line := scanner.Bytes()
+		if i == maxLines {
+			break
+		}
+		truncdata.Write(line)
```

Comment:
This implementation is a bit wasteful, though. We can count lines and bytes very efficiently by iterating over the input bytes, without any allocation at all. The returned slice uses the same array of the input.

Also, the logic seems reversed. We want the end of the buffer, not the start.

For inspiration, here is a function copied out of Spread that does something similar, although without size limitation. It's also a bit more complex because it looks for range of lines instead of just a fixed number. In other words, it displays 10 to 13 lines to avoid cutting a single line. If there are 14, it cuts 4 lines and displays 10. If there are 13, it displays all 13 lines.

No need for the extra complexity if you don't want to go there, though.
```
func tail(output []byte) []byte {
        display := 10
        min := display
        max := display + 3
        mark := 0
        for i := len(output) - 1; i >= 0; i-- {
                if output[i] != '\n' {
                        continue
                }

                min--
                max--

                if min == 0 {
                        mark = i + 1
                        continue
                }
                if max == 0 {
                        return output[mark:]
                }
        }
        return output
}
```